### ISSUE TITLE
\zlabel is ignored inside multline environment

### ISSUE THREAD

**Issue by kia999** [ho-tex/zref#4]
I noted that \zlabel is ignored inside multline environment, it does not write \zref@newlabel to aux file. Here is MWE:

```
\documentclass[a4paper]{article}
\usepackage{amsmath}
\usepackage[user]{zref}


\begin{document}

\begin{equation}
\zlabel{eq:01}
E=mc^{2}
\end{equation}

\begin{multline}
\zlabel{eq:02}
  W = \int \vec{\xi}\cdot \vec{F}(\vec{\xi}) \,d^{3}x 
    =
   \\
   \iint \ldots 
\end{multline}

First Eq.~\zref{eq:01}, second Eq.~\zref{eq:02}.
\end{document}

```

**Comment by u-fischer** [comment]
Yes. But I don't think that zref can do much here. The amsmath environments take a lot trouble to save and restore \label when processing the content but ignore \zlabel. So amsmath would need a change. 

You can get your example more or less working with the following. But it could have bad side effects if you have \label commands in other places:

~~~~
\documentclass[a4paper]{article}

\usepackage[user]{zref}

\makeatletter \def\label{\zref@wrapper@babel\zref@label}\makeatother
\usepackage{amsmath}


\begin{document}

\begin{equation}
\zlabel{eq:01}
E=mc^{2}
\end{equation}

\begin{multline}
\label{eq:02}
  W = \int \vec{\xi}\cdot \vec{F}(\vec{\xi}) \,d^{3}x
    =
   \\
   \iint \ldots
\end{multline}

First Eq.~\zref{eq:01}, second Eq.~\zref{eq:02}.
\end{document}
~~~~

**Comment by kia999** [comment]
@u-fischer : No, it does not work.

**Comment by u-fischer** [comment]
Yes, it does. I tested it on a current texlive.

**Comment by kia999** [comment]
I tested it on current MikTeX. Отправлено из приложения "Почта" для Windows 10 От: Ulrike FischerОтправлено: 6 февраля 2020 г. в 17:42Кому: ho-tex/zrefКопия: Igor A. Kotelnikov; AuthorТема: Re: [ho-tex/zref] \zlabel is ignored inside multline environment (#4) Yes, it does. I tested it on a current texlive.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by davidcarlisle** [comment]
@kia999  your last comment appears scrambled (it just shows as CSS?) could you clarify your "not work" comment by showing the log, or saying what happens?

**Comment by u-fischer** [comment]
@kia999 Works fine for me on miktex too.

**Comment by kia999** [comment]
In my example zref is loaded after amsmath, and \zref prints ?? in \zlabel is inside multline. Log file reports that label is not found.

In example given bu Ulrike Fischer zref goes before amsmath and \zref works fine.

**Comment by davidcarlisle** [comment]
@u-fischer  I suppose it might be possible for `\zlabel` to check if `\label` is `\label@in@display`  (that is check if it's in the scope of a local amsmath alignment redefinition of `\label`) and if so adapt itself accordingly, perhaps...

**Comment by u-fischer** [comment]
@davidcarlisle  the problem isn't so much that zlabel isn't executed, but that even if one force the execution it doesn't get the right values to save them to the aux file. E.g.

~~~
\documentclass[a4paper]{article}
\usepackage{amsmath}
\usepackage[user]{zref}

\begin{document}

\makeatletter
\renewcommand*\zlabel{%
  \zref@wrapper@babel\zref@label
}%

\begin{multline}
%\def\@currentlabel{\theequation}
\zlabel{eq:02}
  W
\end{multline}

Eq.~\zref{eq:02}.
\end{document}

~~~

without the \@currentlabel definition the default is empty `\default{}`, and with it it is 0.
